ntp_operator.py

```diff
@@ -127,39 +140,78 @@ def _init_operator(self, idname: str, label: str) -> None:
         idname (str): name for the operator
         label (str): appearence inside Blender
         """
-        self._write(f"class {self._class_name}(bpy.types.Operator):\n")
-        self._write(f"\tbl_idname = \"object.{idname}\"\n")
-        self._write(f"\tbl_label = \"{label}\"\n")
-        self._write("\tbl_options = {\'REGISTER\', \'UNDO\'}\n")
-        self._write("\n")
-
-    def _is_outermost_node_group(self, level: int) -> bool:
-        if self.mode == 'ADDON' and level == 2:
-            return True
-        elif self.mode == 'SCRIPT' and level == 0:
-            return True
-        return False
-
-    def _process_group_node_tree(self, node: Node, node_var: str, level: int,
-                                 inner: str) -> None:
+        self._write(f"class {self._class_name}(bpy.types.Operator):", "")
+        self._write(f"\tbl_idname = \"object.{idname}\"", "")
+        self._write(f"\tbl_label = \"{label}\"", "")
+        self._write("\tbl_options = {\'REGISTER\', \'UNDO\'}", "")
+        self._write("")
+
+    def _topological_sort(self, node_tree: NodeTree) -> list[NodeTree]:
+        """
+        Perform a topological sort on the node graph to determine dependencies 
+        and which node groups need processed first
+
+        Parameters:
+        node_tree (NodeTree): the base node tree to convert
+
+        Returns:
+        (list[NodeTree]): the node trees in order of processing
+        """
+        if isinstance(node_tree, bpy.types.CompositorNodeTree):
+            group_node_type = 'CompositorNodeGroup'
+        elif isinstance(node_tree, bpy.types.GeometryNodeTree):
+            group_node_type = 'GeometryNodeGroup'
+        elif isinstance(node_tree, bpy.types.ShaderNodeTree):
+            group_node_type = 'ShaderNodeGroup'
+        
+        visited = set()
+        result: list[NodeTree] = []
+
+        def dfs(nt: NodeTree) -> None:
+            """
+            Helper function to perform depth-first search on a NodeTree
+
+            Parameters:
+            nt (NodeTree): current node tree in the dependency graph
+            """
+            if nt not in visited:
+                visited.add(nt)
+                for group_node in [node for node in nt.nodes
+                                   if node.bl_idname == group_node_type]:
+                    if group_node.node_tree not in visited:
+                        dfs(group_node.node_tree)
+                result.append(nt)
+        
+        dfs(node_tree)
+
+        return result
+
+    def _process_group_node_tree(self, node: Node) -> None:
```

Comment:
Shouldn't be a special function for the group node anymore. Add a `ST.NODE_TREE` type, update the settings, and adjust the `set_settings_default()` function. This'll allow for more general node tree settings anyways in case Blender ever decides to use it in more nodes